lib/dialects/oracle/connection-manager.js

```diff
@@ -0,0 +1,197 @@
+      errorToThrow += '\n from tnsnames.ora : (HOST = mymachine.example.com)(PORT = 1521)(SERVICE_NAME = orcl)'; 
+      throw new Error(errorToThrow);
+    }
+
+    if (!config.host || config.host.length === 0) {
+      throw new Error('You have to specify the host');
+    }
+
+    //The connectString has a special format, we check it
+    //ConnectString format is : host:[port]/service_name
+    if (config.database.indexOf('/') === - 1) {
+      
+      let connectString = config.host;
+
+      if (config.port && config.port !== 0) {
+        connectString += `:${config.port}`;
+      } else {
+        connectString += ':1521'; //Default port number
+      }
+      connectString += `/${config.database}`;
+      config.database = connectString;
+    }
+  }
+
+  // Expose this as a method so that the parsing may be updated when the user has added additional, custom types
+  _refreshTypeParser(dataType) {
+    parserStore.refresh(dataType);
+  }
+
+  _clearTypeParser() {
+    parserStore.clear();
+  }
+
+  connect(config) {
+    const self = this;
+    return new Promise((resolve, reject) => {
+
+      const connectionConfig = {
+        user: config.username,
+        host:  config.host,
+        port : config.port,
+        database : config.database,
+        password: config.password,
+        externalAuth : config.externalAuth,
+        stmtCacheSize : 0
+      };
+
+      //We check if there dialect options
+      if ('dialectOptions' in config) {
+        const dialectOptions = config.dialectOptions;
+
+        //If stmtCacheSize is defined, we set it
+        if (dialectOptions && 'stmtCacheSize' in dialectOptions) {
+          connectionConfig.stmtCacheSize = dialectOptions.stmtCacheSize;
+        }
+      }
+
+
+      //Check the config object
+      self.checkConfigObject(connectionConfig);
+
+      //We assume that the database has been correctly formed
+      connectionConfig.connectString = connectionConfig.database;
+
+      if (config.dialectOptions) {
+        Object.keys(config.dialectOptions).forEach(key => {
+          connectionConfig[key] = config.dialectOptions[key];
+        });
+      }
+
+      return self.lib.getConnection(connectionConfig).then(connection => {
+        //TODO Oracle - connection pooling
+        //Not relevant, node-oracledb considers it if multiple connections are opened / closed; while testing, a few connections are created and closed.
+        //We change the session NLS_COMP and NLS_SORT to allow search case insensitive (http://stackoverflow.com/questions/5391069/case-insensitive-searching-in-oracle)
+        const alterSessionQry = "BEGIN EXECUTE IMMEDIATE 'ALTER SESSION SET NLS_COMP=LINGUISTIC'; EXECUTE IMMEDIATE 'ALTER SESSION SET NLS_SORT=BINARY_CI'; END;";
+        return connection.execute(alterSessionQry).then(() => {
+          return connection.commit().then(() => {
```

Comment:
`ALTER SESSION` SQL statements don't require a `COMMIT` as it is not a DML operation.